src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -187,17 +189,31 @@
             mutex = DO_NOT_VALIDATE_FILTERED_RECORDS)
     Boolean VALIDATE_GVCF = false;
 
+    /**
+     * Fail if GVCF contains positions that overlap. Overlapping variants are allowed because they are assumed to be on
+     * separate haplotypes.
+     */
+    @Argument(fullName = "fail-gvcf-on-overlap",
+            shortName = "no-overlaps",
+            doc = "Fail if GVCF contains positions that overlap (except two variants overlapping)",
+            optional = true,
+            mutex = DO_NOT_VALIDATE_FILTERED_RECORDS)
```

Comment:
My only other comment is that you should probably test this. It shouldn't be too hard to generate a tiny file with just 2 slightly overlapping ref blocks and assert that the file crashes successfully... you could also try enabling this on some of our other tests...